cordova-lib/src/cordova/prepare.js

```diff
@@ -56,7 +56,7 @@ function prepare(options) {
     var hooksRunner = new HooksRunner(projectRoot);
     return hooksRunner.fire('before_prepare', options)
     .then(function(){
-        return restore.installPlatformsFromConfigXML(options.platforms);
+        return restore.installPlatformsFromConfigXML(options.platforms, { searchpath : options.searchpath });
```

Comment:
I don't see this searchpath ption being used by [the downstream function]( https://github.com/apache/cordova-lib/blob/master/cordova-lib/src/cordova/restore-util.js#L35 ).